shardingsphere-sql-parser/shardingsphere-sql-parser-engine/src/main/java/org/apache/shardingsphere/sql/parser/api/parser/SQLParserEngine.java

```diff
@@ -64,7 +64,7 @@ public ParseTree parse(final String sql, final boolean useCache) {
     private ParseTree parse(final String sql) {
         ParseASTNode result = twoPhaseParse(sql);
         if (result.getRootNode() instanceof ErrorNode) {
-            throw new SQLParsingException(String.format("Unsupported SQL of `%s`", sql));
+            throw new SQLParsingException(String.format("Unsupported SQL of `%s`", sql).replaceAll("%", "%%"));
```

Comment:
Why is `replaceAll("%", "%%")` needed?

Reply:
> Why is `replaceAll("%", "%%")` needed?

@tristaZero When user executes a SQL containing `%` and the SQL parsing is abnormal, for example: `SET PASSWORD FOR'bob'@'%.example.org' = PASSWORD('auth_string');`. Since the `%` symbol is a reserved character, an `UnknownFormatConversionException` will occur when the `String.format()` method is executed.

In order to allow users to see the real prompt, such as: `Unsupported SQL of xxxStatement`, we need to escape the `%` symbol, that is, replace it with the `%%` symbol.